pip_audit/_service/interface.py

```diff
@@ -87,16 +88,21 @@ class VulnerabilityResult:
     A human-readable description of the vulnerability.
     """
 
-    fix_versions: List[Version]
+    fix_versions: list[Version]
     """
     A list of versions that can be upgraded to that resolve the vulnerability.
     """
 
-    aliases: Set[str]
+    aliases: set[str]
     """
     A set of aliases (alternative identifiers) for this result.
     """
 
+    published: datetime | None = None
```

Comment:
Does this work in Python 3.7? I know that PEP 585 (`dict[...]` instead of `Dict[...]`) does with `future`, but PEP 604 doesn't mention similar support.

Reply:
+ PEP 585 is introduced in Python 3.9 and adds support for generic built-ins. 
+ PEP 604 is introduced in Python 3.10 and adds support for union operator on types.
+ PEP 563 is introduced in Python and adds `from __future__ import annotations` making all annotations lazy.
+ Both PEP 585 and PEP 604 can be used in Python 3.7 and 3.9 in combination with PEP 563 because annotations aren't evaluated at runtime.

Reply:
> PEP 563 is introduced in Python 3.7 and adds from __future__ import annotations making all annotations lazy.

TIL, thank you!